server/src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -305,9 +304,13 @@ public void ensureUUID() {
 
     /**
      * @param uuid the UUID of this consumer.
+     * @return this consumer.
      */
-    public void setUuid(String uuid) {
-        this.uuid = uuid;
+    public Consumer setUuid(String uuid) {
+        if (this.uuid == null) {
```

Comment:
What's the intent behind this? Generally we want to allow this to be changed at any point. Is there something in particular that requires this with the update?

Reply:
I thought we should not change the uuid once set.

Reply:
I'll correct this one.

Reply:
Corrected.